activerecord/lib/active_record/model_schema.rb

```diff
@@ -478,6 +478,9 @@ def load_schema
             load_schema!
 
             @schema_loaded = true
+          rescue
+            @columns_hash = nil # If the schema loading failed half way through, we must reset the state.
```

Comment:
I'm actually tempted to call `reload_schema_from_cache` here.

Reply:
:+1:

Reply:
Done.